public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -1339,7 +1340,23 @@ export class DashboardModel implements TimeModel {
     return this.originalDashboard;
   }
 
-  hasAngularPlugins(): boolean {
+  async hasAngularPlugins(): Promise<boolean> {
+    // Ensure all panel plugins are loaded before checking if they are Angular plugins
```

Comment:
Do we need to load the plugins to know if they are angular plugins? I thought the information was already available in `config.panels`. Couldn't we use that?

Reply:
You're right, I wanted to use `panel.isAngularPlugin()` across all warnings for consistency, but by accessing `grafanaBootData` makes the PR much simpler, plus it solves the issue about having to load all plugins on the first page load you were talking about in your other comment.

I have pushed some changes so it uses `grafanaBootData` instead